### PR TITLE
Generate K8s Injection Code

### PR DESCRIPTION
**Pull request by dprotaso**

### CODE REVIEW

**Comment by dprotaso** [comment]
Going to stage draft PRs in other repos

File: codegen/cmd/injection-gen/generators/client.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func Get(ctx context.Context) {{.clientSetInterface|raw}} {
 	untyped := ctx.Value(Key{})
 	if untyped == nil {
 		{{.loggingFromContext|raw}}(ctx).Fatalf(
-			"Unable to fetch %T from context.", ({{.clientSetInterface|raw}})(nil))
+			"Unable to fetch {{.clientSetInterface}} from context.")
```
Comment (mattmoor) [inline codegen/cmd/injection-gen/generators/client.go]: This can now be `.Fatal` (everywhere)

Comment (dprotaso) [reply codegen/cmd/injection-gen/generators/client.go]: true

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -39,6 +39,23 @@ ${REPO_ROOT_DIR}/hack/generate-knative.sh "injection" \
...
+OUTPUT_PKG="knative.dev/pkg/injection/kube" \
+VERSIONED_CLIENTSET_PKG="k8s.io/client-go/kubernetes" \
+EXTERNAL_INFORMER_PKG="k8s.io/client-go/informers" \
+  ${REPO_ROOT_DIR}/hack/generate-knative.sh "injection" \
+    k8s.io/client-go \
+    k8s.io/api \
+    "apps:v1 autoscaling:v1,v2beta1 batch:v1,v1beta1 core:v1 rbac:v1" \
+    --go-header-file ${REPO_ROOT_DIR}/hack/boilerplate/boilerplate.go.txt
+
+OUTPUT_PKG="knative.dev/pkg/injection/apiextensions" \
```
Comment (mattmoor) [inline hack/update-codegen.sh]:
```suggestion
OUTPUT_PKG="knative.dev/pkg/client/injection/apiextensions" \
```

File: hack/update-codegen.sh [hunk 3]
```diff
@@ -39,6 +39,23 @@ ${REPO_ROOT_DIR}/hack/generate-knative.sh "injection" \
   "istio:v1alpha3 istio/authentication:v1alpha1" \
   --go-header-file ${REPO_ROOT_DIR}/hack/boilerplate/boilerplate.go.txt
 
+OUTPUT_PKG="knative.dev/pkg/injection/kube" \
```
Comment (mattmoor) [inline hack/update-codegen.sh]:
```suggestion
OUTPUT_PKG="knative.dev/pkg/client/injection/kube" \
```

File: codegen/cmd/injection-gen/generators/client.go (hunk 1 above)
Comment (dprotaso) [reply codegen/cmd/injection-gen/generators/client.go]: Any thoughts to panic'ing? If tests aren't setup incorrectly then it just exits.

**Comment by dprotaso** [comment]
- [ ] Sample Controller - https://github.com/knative/sample-controller/pull/97

**Comment by dprotaso** [comment]
looks like we have redundant factories

**Comment by dprotaso** [comment]
ie. 
These two are identical

https://github.com/knative/serving/blob/master/pkg/client/injection/informers/autoscaling/factory/autoscalingfactory.go

https://github.com/knative/serving/blob/master/pkg/client/injection/informers/serving/factory/servingfactory.go

**Comment by mattmoor** [comment]
@dprotaso LMK when this is RFAL

**Comment by dprotaso** [comment]
@mattmoor rebased please take a look

**Comment by n3wscott** [comment]
awesome

**Pull request merged**